File: solutions/c/binary-search/1/binary_search.c
#include "binary_search.h"

const int *binary_search(int value, const int *arr, size_t length){
  if(length == 0) return NULL;
  int low = 0;
  int high = length;
  while(high >= low){
    int mid = (low + high) / 2;
    if(arr[mid] == value){
      return &arr[mid];
    }
    else if(value < arr[mid]) high = mid - 1;
    else low = mid + 1;
  }
  return NULL;
}

File: solutions/csharp/binary-search/1/BinarySearch.cs
public static class BinarySearch
{
    public static int Find(int[] input, int value)
    {
        int high = input.Length - 1;
        int low = 0;
        while(low <= high){
            int mid = (high + low) / 2;
            if(input[mid] == value) return mid;
            else if(input[mid] < value) low = mid + 1;
            else high = mid - 1;
        }
        return -1;
    }
}